omod/src/test/java/org/openmrs/module/reportingrest/web/resource/ReportRequestResourceTest.java

```diff
@@ -60,6 +72,93 @@ public void setUp() throws Exception {
 	private ReportRequestResource getResource() {
 		return (ReportRequestResource) Context.getService(RestService.class).getResourceBySupportedClass(ReportRequest.class);
 	}
+
+	@Test
+	public void testCreateWithReportDefinitionAsParameters() throws Exception {
+		String reportDefinitionJson = "{"
+				+ "\"parameterizable\":{"
+				+ "   \"uuid\":\"" + REPORT_DEFINITION_UUID + "\""
+				+ "},"
+				+ "\"parameterMappings\":{"
+				+ "   \"endDate\":\"2017-01-31\","
+				+ "   \"startDate\":\"2017-01-01\"}"
+				+ "}";
+
+		String reportRequestJson = "{\n" +
+				"  \"status\": \"REQUESTED\",\n" +
+				"  \"priority\": \"NORMAL\",\n" +
+				"  \"reportDefinition\":" + reportDefinitionJson + "," +
+				"  \"renderingMode\": \"org.openmrs.module.reporting.report.renderer.CsvReportRenderer\"\n" +
+				"}";
+
+		SimpleObject properties = SimpleObject.parseJson(reportRequestJson);
+		RequestContext context = new RequestContext();
+		context.setRepresentation(Representation.DEFAULT);
+		SimpleObject response = (SimpleObject) getResource().create(properties, context);
+		assertNotNull(response.get("uuid"));
+		assertEquals(response.get("status"), ReportRequest.Status.REQUESTED);
+		assertEquals(response.get("priority"), ReportRequest.Priority.NORMAL);
+		SimpleObject resultObject = (SimpleObject) response.get("renderingMode");
+		String rendererType = (String) resultObject.get("rendererType");
+		assertEquals(rendererType, "org.openmrs.module.reporting.report.renderer.CsvReportRenderer");
+
+		ReportRequest request = getResource().getByUniqueId((String) response.get("uuid"));
+		assertThat(request.getReportDefinition().getParameterizable().getUuid(), is(REPORT_DEFINITION_UUID));
+		assertThat(request.getReportDefinition().getParameterMappings().containsKey("startDate"), is(true));
+		assertThat(request.getReportDefinition().getParameterMappings().containsKey("endDate"), is(true));
```

Comment:
Yes, this is what I wanted to see.  Can you also test that the parameter values are properly set?  Basically, we want to test that the reportrequest resource is successfully able to create a ReportRequest with all of the associated values.  So any value we are passing in we need to test whether it actually gets correctly assigned.

Reply:
@meaton updated the test case with the values that we pass as part of requestBody

Reply:
I think that @mseaton is saying that he expects to see a test against the value of "2017-01-31" (instead of just `containsKey`.

Reply:
@djazayeri Addressed your comment and changed the test case accordingly.